components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/DefaultIdentityProviderImpl.java

```diff
@@ -87,6 +91,54 @@ public boolean isValidRole(String roleName) {
     }
 
     @Override
+    public List<String> getRoleIdsOfUser(String userId) throws IdentityProviderException {
+        List<String> roleIds = new ArrayList<>();
+        SCIMUser scimUser = scimServiceStub.getUser(userId);
+        if (scimUser != null) {
+            List<SCIMUser.SCIMUserGroups> roles = scimUser.getGroups();
+            if (roles != null) {
+                roles.forEach(role -> roleIds.add(role.getValue()));
+            }
+        } else {
+            String errorMessage = "User id " + userId + " does not exist in the system.";
+            log.error(errorMessage);
+            throw new IdentityProviderException(errorMessage, ExceptionCodes.USER_DOES_NOT_EXIST);
+        }
+        return roleIds;
+    }
+
+    @Override
+    public String getRoleId(String roleName) throws IdentityProviderException {
+        Response role = scimServiceStub.searchGroups(FILTER_PREFIX + roleName);
+        String roleId = null;
+        if (role.status() == 200) {
+            String responseBody = role.body().toString();
```

Comment:
ah okey. So if the search role isn't there. It will return with some other status code right? Then it should be fine. Do we require an else statement to handle other status codes?